### PR TITLE
Adding Addon Templating Engine + Web App Routing

### PR DESCRIPTION
**Pull request by imiller31**
This PR aims to serve as the new `update` command and implements more idiomatic templating and patterns.

With this change, arbitrary CRDs can be generated with user inputs. This templating strategy should unify with Builders and DeployTypes in the future.

### CODE REVIEW

**Review by davidgamero — CHANGES_REQUESTED** [review]

File: pkg/config/addon_config.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package config
```
[the remark this answers is not in the document]
Comment (imiller31) [reply pkg/config/addon_config.go:1]: sync'd offline

**Review by davidgamero — APPROVED** [review]
LGTM

**Comment by davidgamero** [comment]
yikes got the flag issue on the integration tests again

**Comment by davidgamero** [comment]
rebased to #136 due to signed commits issue

**Pull request closed without merging**